tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,182 @@
+    t = str(t)
+    s_len, t_len = len(s), len(t)
+    s_index, t_index = 0, 0
+    while (s_index < s_len and t_index < t_len):
+        if (s[s_index] == t[t_index]):
+            s_index += 1
+        t_index += 1
+    return s_index == s_len
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [GnuTLS, OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_preference", FRAGMENT_PREFERENCE, ids=get_parameter_name)
+def test_s2n_buffered_send_server(managed_process, cipher, other_provider, provider, protocol, certificate, buffer_size, fragment_preference):
+    # Communication Timeline
+    # Client [S2N|OpenSSL|GnuTLS]  | Server [S2N]
+    # Handshake                    | Handshake
+    #  Handshake finish indicated
+    #  by the client send marker
+    # Send Server Send Marker      | Receive the Server Send Marker "CLIENTSENT"
+    #                              | Send Data Bytes to Client (stresses the send buffer - center of test)
+    #                              | Send Client Close Marker "SERVERSENT"
+    # Close                        | Close
+    
+    # for the corresponding provider, these strings appear when on stdout when the handshake is finished
+    starting_client_send_markers = {
+        GnuTLS: "Handshake was completed",
+        OpenSSL: "write to",
+        S2N: "ready"
+    }
+    starting_client_send_marker = starting_client_send_markers[other_provider]
+
+    starting_server_send_marker = "YTREWQ"
+    client_inital_app_data = to_bytes(starting_server_send_marker)
+    client_close_marker = server_sent_final = "QWERTY"
+    data_bytes_server = data_bytes(SEND_DATA_SIZE) + to_bytes(server_sent_final)
+
+    port = next(available_ports)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        data_to_send=client_inital_app_data,
+        insecure=True,
+        protocol=protocol)
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=data_bytes_server,
+        insecure=True,
+        protocol=protocol,
+        key=certificate.key,
+        cert=certificate.cert,
+        extra_flags=['--buffered-send', buffer_size] +
+            ([] if fragment_preference is None else [fragment_preference]))
+
+    server = managed_process(provider, server_options, timeout=60, send_marker=[starting_server_send_marker])
+    client = managed_process(other_provider, client_options, timeout=60,
```

Comment:
Do the timeouts need to be this big?

Reply:
No they could probably be smaller.

Reply:
Indeed it can, using default value of 5.